From: Richard Acayan <mailingradian@gmail.com>
Date: Mon, 31 Oct 2022 17:40:58 -0400
Subject: [PATCH] arm64: dts: qcom: sdm670-google: add nfc support

The NFC driver can at least detect an NFC tag.

Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 arch/arm64/boot/dts/qcom/sdm670-google-common.dtsi | 35 ++++++++++++++++++++++
 1 file changed, 35 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sdm670-google-common.dtsi b/arch/arm64/boot/dts/qcom/sdm670-google-common.dtsi
index 4d535a9..f970da8 100644
--- a/arch/arm64/boot/dts/qcom/sdm670-google-common.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm670-google-common.dtsi
@@ -504,6 +504,28 @@ zap-shader {
 	};
 };
 
+&i2c3 {
+	clock-frequency = <400000>;
+	status = "okay";
+
+	nfc@28 {
+		/*
+		 * The interface is actually NQ (PN557) but this should work
+		 * fine until we add embedded secure element.
+		 */
+		compatible = "nxp,nxp-nci-i2c";
+		reg = <0x28>;
+
+		interrupts-extended = <&tlmm 44 IRQ_TYPE_LEVEL_HIGH>;
+
+		enable-gpios = <&tlmm 12 GPIO_ACTIVE_HIGH>;
+		firmware-gpios = <&tlmm 43 GPIO_ACTIVE_HIGH>;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&nfc_default>;
+	};
+};
+
 &i2c9 {
 	clock-frequency = <100000>;
 	status = "okay";
@@ -768,6 +790,19 @@ drv2624_default_state: drv2624-default-state {
 		bias-disable;
 		output-low;
 	};
+
+	nfc_default: nfc-default-state {
+		/*
+		 * gpio12: enable
+		 * gpio43: firmware
+		 * gpio44: irq
+		 * gpio116: ese
+		 */
+		pins = "gpio12", "gpio43", "gpio44", "gpio116";
+		function = "gpio";
+		drive-strength = <2>;
+		bias-pull-down;
+	};
 };
 
 &uart6 {
